force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +161,35 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        List <Account> acc = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        if (acc.size() == 0) {
+           Account account = new Account(Name = accountName);
+           acc.add(account);
+            upsert acc;
+        }
+        Account acct = acc[0];
+        List<Opportunity> existingOpportunities = [SELECT Id, Name, AccountId FROM Opportunity WHERE Name IN :oppNames AND AccountId = :acct.Id];
+       List <Opportunity> oppList = new List <Opportunity>();
+        for (String oppName : oppNames) {
+            Opportunity opp = new Opportunity();
+        for (Opportunity existingOpp : existingOpportunities) {
```

Comment:
When you find yourself nesting for loops, this is typically a sign that there's a more efficient way to write your code. Also, watch your indentation. The second for loop should be indented:
```suggestion
        for (String oppName : oppNames) {
            Opportunity opp = new Opportunity();
            for (Opportunity existingOpp : existingOpportunities) {
```